editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/flag/FlagMapFragment.java

```diff
@@ -88,20 +90,26 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
 
     @Override
     public boolean onOptionsItemSelected (MenuItem item) {
+        FragmentManager manager = getActivity().getSupportFragmentManager();
+        FragmentTransaction transaction = manager.beginTransaction();
         //Using a switch here so that eventually, when we add more options in the toolbar, this function can handle multiple possible selections
         switch (item.getItemId()) {
-            //When the "List View" button is selected, swaps this fragment out for the flag list fragment
             case R.id.action_to_listview:
-                FragmentManager manager = getActivity().getSupportFragmentManager();
-                FragmentTransaction transaction = manager.beginTransaction();
+                //When the "List View" button is selected, swaps this fragment out for the flag list fragment
                 transaction.replace(R.id.container, new FlagListFragment());
                 transaction.commit();
                 break;
             case R.id.action_edit:
-                //TODO: insert code to edit flag here
+                //TODO: ensure this is the right fragment to go to
+                transaction.replace(R.id.container, new DescriptionPanelFragment());
+                transaction.commit();
                 break;
             case R.id.action_delete:
-                //TODO: insert code to delete flag here
+                //Deletes flag from room database
+                mViewModel.deleteFlag();
+                //Returns to FlagListFragment since there is now no flag to interact with in FlagMapFragment
+                transaction.replace(R.id.container, new FlagListFragment());
```

Comment:
I think FlagMapFragment still has an empty terrain even if there's no flag, so we can safely return there.